### PR TITLE
Add /clear to PMs

### PR DESCRIPTION
**Pull request by sparkychildcharlie**
In response to [this suggestion that I find useful and interesting][1], I added /clear to the PM box.  This does:
- closes the pm box
- flushes out chat history and the inner HTML.

(PS: Please let me know if you want this as a regular chatroom based command as well, I can __totally__ try to add that too!)

[1]: http://www.smogon.com/forums/threads/suggestions.3534365/page-49#post-7230510

### CODE REVIEW

File: js/client-mainmenu.js [hunk 1]
```diff
@@ -326,6 +326,9 @@
 					delete app.ignore[userid];
 					$chat.append('<div class="chat">User ' + userid + ' no longer ignored.</div>');
 				}
+			} else if (text.toLowerCase() === '/clear') {
+				this.closePM(userid);
+				this.$pmBox.find('.pm-window-' + userid).find('.inner').empty();
```
Comment (Slayer95) [inline js/client-mainmenu.js]: This could be just ``$pmWindow.find('.inner').empty()`` afaict.

Comment (sparkychildcharlie) [inline js/client-mainmenu.js]: ah thanks! Sorry, I'm not as familiar with client... yet!

Comment (sparkychildcharlie) [inline js/client-mainmenu.js]: (i used ``$chat.empty()``)

**Comment by Slayer95** [comment]
It's not intuitive for ``/clear`` to also close the PM.

> as a regular chatroom based command

To be *clear*, this is already a client-side command for chat rooms, and it doesn't close the room.

**Comment by sparkychildcharlie** [comment]
@Slayer95 would you prefer another name, or for this just to __not__ close the pm?

**Comment by Slayer95** [comment]
Also,

> flushes out chat history

I don't see where this is done? Perhaps you forgot to commit it?

**Comment by sparkychildcharlie** [comment]
doesn't ``$chat.empty()`` do that? This is what is used in the chatroom command /clearpms (of which this is a single instance of what that does) and after using it, reopening the pm box will not have any of the previous proceedings showing.

Or is there something very important I'm overlooking?

**Comment by Slayer95** [comment]
Oh, *that* history. I thought you meant the history that is browsed with Up Arrow / Down Arrow, which ``/clearpms`` doesn't flush either. This would be done with ``delete this.chatHistories[userid]``.

**Comment by sparkychildcharlie** [comment]
I'll wipe that too then i guess.

**Comment by Zarel** [comment]
...no, don't wipe up/down history :|

**Comment by Zarel** [comment]
Also, I thought we agreed not to close the PM?

**Comment by sparkychildcharlie** [comment]
Sorry Zarel, I didn't notice the edit in that post.   This just clears the HTML in the pm box then.

**Comment by Zarel** [comment]
Looks good. Thanks!

**Pull request merged**